app/models/instrument_for_cart.rb

```diff
@@ -0,0 +1,26 @@
+class InstrumentForCart < ProductForCart
+
+  def purchasable_by?(acting_user, session_user)
+    # If an instrument is not purchasable, we always redirect. Hence, if super
+    # doesn’t set an error path due to some specific error, we set the error path
+    # to the facility path.
+    super.tap do |is_purchasable|
```

Comment:
This tap still feels confusing. Usually I only like tap if we're doing builder-y things.

```ruby
is_purchasable = super
@error_path ||= url_helpers.facility_path(product.facility) unless is_purchasable
is_purchasable
```